Bottom: 961db4b57229694823ccfcc79edd8cff5bdca892
Top:    48db3f194311d8b78b8ca9ab8a76ddc6bb750bc3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 17:44:49 +0100

TEMP


---

diff --git a/src/draw.c b/src/draw.c
index f12020d..d096377 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -102,6 +102,7 @@ static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
 static void ClearPad (PadTypePtr, bool);
 static void DrawHole (PinTypePtr);
 static void DrawMask (int side, BoxType *);
+static void DrawPaste (int side, BoxType *);
 static void DrawRats (BoxType *);
 static void DrawSilk (int side, const BoxType *);
 static int via_callback (const BoxType * b, void *cl);
@@ -704,6 +705,26 @@ DrawMask (int side, BoxType * screen)
     }
 }
 
+/* ---------------------------------------------------------------------------
+ * draws solder paste layer for a given side of the board
+ */
+static void
+DrawPaste (int side, BoxType *drawn_area)
+{
+  gui->set_color (Output.fgGC, PCB->ElementColor);
+  ALLPAD_LOOP (PCB->Data);
+  {
+    if (ON_SIDE (pad, side) && !TEST_FLAG (NOPASTEFLAG, pad) && pad->Mask > 0)
+      {
+        if (pad->Mask < pad->Thickness)
+          DrawPadLowLevel (Output.fgGC, pad, true, true);
+        else
+          DrawPadLowLevel (Output.fgGC, pad, false, false);
+      }
+  }
+  ENDALL_LOOP;
+}
+
 static void
 DrawRats (BoxTypePtr drawn_area)
 {
diff --git a/src/misc.c b/src/misc.c
index 02a883f..d7aad5d 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -600,6 +600,22 @@ IsLayerGroupEmpty (int num)
   return true;
 }
 
+static bool
+IsPasteEmpty (int side)
+{
+  bool paste_empty = true;
+  ALLPAD_LOOP (PCB->Data);
+  {
+    if (ON_SIDE (pad, side) && !TEST_FLAG (NOPASTEFLAG, pad) && pad->Mask > 0)
+      {
+        paste_empty = false;
+        break;
+      }
+  }
+  ENDALL_LOOP;
+  return paste_empty;
+}
+
 /* ---------------------------------------------------------------------------
  * gets minimum and maximum coordinates
  * returns NULL if layout is empty
